object.go

```diff
@@ -757,6 +757,35 @@ func (o *Object) ValueNotEqual(key string, value interface{}) *Object {
 	return o.NotValueEqual(key, value)
 }
 
+// Iter returns a new map of Value
+//
+// Example:
+//
+// object := newObject(chain, map[string]interface{})
+//
+//	for n, val := range object.Iter() {
+//		    val.Equal(object[n])
+//		}
```

Comment:
* Please fix indentation
* Let's rename `n` to `key` and `val` to `value`
* Let's provide a valid and complete example similar to one in `Array.Iter`